subsys/bluetooth/host/id.c

```diff
@@ -1407,16 +1451,22 @@ uint8_t bt_read_static_addr(struct bt_hci_vs_static_addr addrs[], uint8_t size)
 		return 0;
 	}
 
-	for (i = 0; i < cnt; i++) {
-		memcpy(&addrs[i], rp->a, sizeof(struct bt_hci_vs_static_addr));
+	valid_id_cnt = ret_id_cnt;
+	for (i = 0; i < ret_id_cnt; i++) {
+		if (!BT_ADDR_IS_STATIC(&rp->a[i].bdaddr)) {
+			valid_id_cnt--;
+			continue;
+		}
+
+		memcpy(&addrs[i], &rp->a[i], sizeof(struct bt_hci_vs_static_addr));
```

Comment:
```suggestion
		(void)memcpy(&addrs[i], &rp->a[i], sizeof(struct bt_hci_vs_static_addr));
```

Conventionally we void cast the results of the std lib mem functions